### DISCUSSION TITLE
VASP5.4.4 + wan90v2 memory issue?

### DISCUSSION THREAD

**Discussion by Dongsheng-Wen** [Chengcheng-Xiao/VASP2WAN90_v2_fix#13] [General]
Hello,
I have a question about compiling/using the vasp and wannier90 codes, but I couldn't find useful resources online. Then I saw this patch and think you might be able to give me some hints. 
I was trying to compile vasp5.4.4 with wannier90v2 interface for SOC calculations. The compiled vasp seemed to be working fine. The executables were working without setting the wannier tags. Then I tested the code with the iron example on your repo. When I turned on the wannier tags to write the outputs .mmn and .amn, vasp stopped with errors like: 
```Program received signal SIGSEGV: Segmentation fault - invalid memory reference.```
The .mmn file was written but .amn was not created. 

For this simple Fe system, my computational power is good enough (512G memory + 128 cpus). It seems to me the memory issure is related to the wannier interface. So I wonder if the patch could solve this problem? 

Thanks,
Dongsheng

**Comment by Chengcheng-Xiao** [comment]
Hi Dongsheng 👋
Yes, this patch will fix your problem. But the problem isn't caused by not having enough memory, it's caused by a bug in the original interface.

Pleas see https://github.com/Chengcheng-Xiao/VASP2WAN90_v2_fix/wiki/what's-wrong-with-the-original-interface%3F#original-symptoms for more insight.

**Comment by Dongsheng-Wen** [comment]
Hi Chengcheng, 
Thank you so much! I was also wondering if this patch works for v3.0.0 and vasp6?

**Reply by Chengcheng-Xiao** [reply]
No, this patch wasn't designed for VASP v6.
VASP v6.2.0+ shouldn't need this patch. (unless you want `.spn` matrix output)
